app/src/main/java/org/jellyfin/mobile/fragment/WebViewFragment.kt

```diff
@@ -24,27 +24,34 @@ import androidx.webkit.WebViewClientCompat
 import androidx.webkit.WebViewFeature
 import kotlinx.coroutines.launch
 import org.jellyfin.apiclient.interaction.ApiClient
-import org.jellyfin.mobile.AppPreferences
 import org.jellyfin.mobile.MainActivity
 import org.jellyfin.mobile.bridge.ExternalPlayer
 import org.jellyfin.mobile.bridge.NativeInterface
 import org.jellyfin.mobile.bridge.NativePlayer
+import org.jellyfin.mobile.controller.ServerController
 import org.jellyfin.mobile.databinding.FragmentWebviewBinding
 import org.jellyfin.mobile.utils.*
+import org.jellyfin.mobile.utils.Constants.FRAGMENT_WEB_VIEW_EXTRA_SERVER_ID
+import org.jellyfin.mobile.utils.Constants.FRAGMENT_WEB_VIEW_EXTRA_URL
 import org.jellyfin.mobile.webapp.WebappFunctionChannel
 import org.json.JSONException
 import org.json.JSONObject
 import org.koin.android.ext.android.inject
 import timber.log.Timber
 import java.io.Reader
 import java.util.*
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
 
 class WebViewFragment : Fragment() {
     val apiClient: ApiClient by inject()
-    private val appPreferences: AppPreferences by inject()
+    private val serverController: ServerController by inject()
     private val webappFunctionChannel: WebappFunctionChannel by inject()
     private val externalPlayer by lazy { ExternalPlayer(this) }
 
+    private var serverId: Long = 0
```

Comment:
Why no lateinit for this one?

Reply:
Cause lateinit doesn't work for primitives.